packages/blocks/README.md

```diff
@@ -190,9 +190,9 @@ optionally merging new attributes and/or replacing its inner blocks.
 
 _Parameters_
 
--   _block_ `Object`: Block instance.
--   _mergeAttributes_ `Object`: Block attributes.
--   _newInnerBlocks_ `?Array`: Nested blocks.
+-   _block_ `WPBlock`: Block instance.
+-   _mergeAttributes_ `WPBlockAttributes`: Block attributes.
+-   _newInnerBlocks_ `[WPBlock[]]`: Nested blocks.
```

Comment:
Is it an array of arrays of WPBlock objects? Or just an array of WPBlock objects? Ditto in line 209

Reply:
```
/**
 * Returns a block object given its type and attributes.
 *
 * @param {string}            name          Block name.
 * @param {WPBlockAttributes} attributes    Block attributes.
 * @param {WPBlock[]}         [innerBlocks] Nested blocks.
 *
 * @return {WPBlock} Block object.
 */

I think something odd is going the document build process, because wrapping a parameter name should to mean that it is optional. As you see in the JSDoc declaration it its an array of WPBlock that is an optional parameter.

Reply:
> The old JSDocs used the syntax ?Type but the type checker marks that as Type|null which isn't the same as Type|undefined.

Yup, that was wrong, good spot.

Is this readme generated? If it's generating `[WPBlock[]]` that's a bug in whatever is generating it.

Reply:
Yes it was generated.

Reply:
I'm not actually sure what syntax that generated doc is supposed to be. It looks JSDoc-inspired but the optional `[]` go around the name, not around the type.

[There's an alternate syntax we could try](https://jsdoc.app/tags-param.html#optional-parameters-and-default-values) that should look like this:

```js
/**
 * @param {WPBlock[]=}         innerBlocks Nested blocks.
 */
```

I think it's the [docgen](https://github.com/WordPress/gutenberg/tree/trunk/packages/docgen) package generating these, so we could dig around there and see if that can be improved. We might just need to prefix the description with `Optional` or something :confused:

Reply:
@sirreal That syntax did not fix the issue.

Reply:
Looks like the `Type=` syntax works. And it no longer looks like it's typing parameters as a tuple of one element.